internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -239,26 +241,49 @@ func (c *container) setupSIFOverlay(img *image.Image, writable bool) error {
 	return nil
 }
 
+// checkOverlay will test if overlay is supported/allowed. On recent ubuntu
+// distribution user namespace can use overlay filesystem, check mount error
+// to determine if we can use it
+func (c *container) checkOverlay() bool {
+	if _, err := c.rpcOps.Mount("none", "/", "overlay", syscall.MS_SILENT, ""); err != nil {
+		// invalid argument error means overlay is supported and is allowed,
+		// if RPC server is in a user namespace, we should get a permission
+		// denied error on systems not allowing use of overlay from user
```

Comment:
```suggestion
		// denied error on systems not allowing the use of overlay from user
```

Reply:
Seems like this whole comment is deleted now, so this is moot I think.